tests/jointcalTestBase.py

```diff
@@ -0,0 +1,311 @@
+        if getattr(self, 'jointcalTask', None) is not None:
+            del self.catalogs
+
+    def _prep_reference_loader(self, center, radius):
+        """
+        Setup an astrometry.net reference loader.
+
+        @param center (afw.coord) The center of the field you're testing on.
+        @param radius (afw.geom.angle) The radius to load objects around center.
+        """
+        refLoader = LoadAstrometryNetObjectsTask(LoadAstrometryNetObjectsConfig())
+        # Make a copy of the reference catalog for in-memory contiguity.
+        self.reference = refLoader.loadSkyCircle(center, radius, filterName='r').refCat.copy()
+
+    def _testJointCalTask_run(self, nCatalogs, relative_error, absolute_error):
+        """Test jointcal.run() on nCatalogs, requiring less than some error (arcsec)."""
+
+        result = self.jointcalTask.run(self.catalogs[:nCatalogs])
+        self.dataRefs = result.dataRefs
+        self.old_wcss = result.old_wcss
+
+        rms_rel, rms_abs = self.compute_rms(self.catalogs[:nCatalogs], fluxlimit=100)
+        self.assertLess(rms_rel, relative_error/arcsec_per_radian)
+        self.assertLess(rms_abs, absolute_error)
+
+    def _testJointCalTask(self, nCatalogs, relative_error, absolute_error):
+        """Test parseAndRun for jointcal on nCatalogs, requiring less than some error (arcsec)."""
+
+        visits = '^'.join(str(v) for v in self.visitList[:nCatalogs])
+        output_dir = os.path.join('.test', self.__class__.__name__)
+        result = jointcal.JointcalTask.parseAndRun(args=[self.input_dir, '--output', output_dir,
+                                                         '--clobber-versions', '--clobber-config',
+                                                         '--doraise',
+                                                         '--id', 'visit=%s'%visits],
+                                                   doReturnResults=True)
+        self.dataRefs = result.resultList[0].result.dataRefs
+        self.old_wcss = result.resultList[0].result.old_wcss
+
+        rms_rel, rms_abs = self.compute_rms(self.dataRefs, fluxlimit=100)
+        self.assertLess(rms_rel, relative_error/arcsec_per_radian)
+        self.assertLess(rms_abs, absolute_error)
+
+    def compute_rms(self, dataRefs, fluxlimit=100):
+        """
+        Match all dataRefs to compute the RMS, for all detections above some flux limit.
+
+        @param dataRefs (list) the dataRefs to do the relative matching between
+        @param fluxlimit (float) SN (flux/fluxSigma) for sources to be included in match.
+
+        @return (float, float) relative, absolute RMS of stars, in radians
+        """
+
+        visit_list = [dataRef.dataId['visit'] for dataRef in dataRefs]
+
+        old_cats = [dataRef.get('src') for dataRef in dataRefs]
+        visitCatalogs = self._make_visit_catalogs(old_cats, visit_list)
+        refSrc = visitCatalogs.values()[0]  # use the first catalog as the relative reference catalog
```

Comment:
Now I get what you mean. In this case it doesn't matter, as the lists are going to be small. Also, I don't want to break python 3.